src/clj/web/tournament.clj

```diff
@@ -124,72 +130,72 @@
               username2 (get-in table [:player2 :name])
               base {:tournament-name (:name data)
                     :tournament-format "standard"
+                    :cobra-link (:cobra-link data)
                     :selected-round (inc selected-round)
                     :table (:table table)
                     :username1 username1
                     :username2 username2
+                    :save-replay save-replays
                     :allow-spectator true}]
           (create-tournament-lobby
             (assoc base :on-close #(create-tournament-lobby
                                      (assoc base :username1 username2 :username2 username1))))))
       round)))
 
 (defn load-tournament
-  [{{user :user} :ring-req
-    {:keys [cobra-link]} :?data
-    client-id :client-id
-    reply-fn :?reply-fn
-    :as msg}]
-  (if (:tournament-organizer user)
-    (let [data (download-cobra-data cobra-link)
-          player-count (count (:players data))
-          player-names (keep :name (:players data))
-          query (into [] (for [username player-names] {:username username}))
-          db-players (find-maps-case-insensitive db "users" {$or query})
-          missing-players (remove #(seq (filter (fn [e] (= (lower-case %) (lower-case e))) (map :username db-players))) player-names)
-
-          players (build-players data)
-          rounds (process-all-rounds data players)]
-      (ws/broadcast-to! [client-id] :tournament/loaded {:data {:players players
-                                                      :missing-players missing-players
-                                                      :rounds rounds
-                                                      :tournament-name (:name data)}})
-      (when reply-fn (reply-fn 200)))
-    (when reply-fn (reply-fn 403))))
+  [{{:keys [cobra-link]} :?data
+    client-id :client-id}]
+  (let [data (download-cobra-data cobra-link)
+        player-names (keep :name (:players data))
+        query (into [] (for [username player-names] {:username username}))
+        db-players (find-maps-case-insensitive db "users" {$or query})
+        found-player-names #(seq (filter (fn [e] (= (lower-case %) (lower-case e))) (map :username db-players)))
+        missing-players (remove found-player-names player-names)
+        players (build-players data)
+        rounds (process-all-rounds data players)]
+    (ws/broadcast-to! [client-id] :tournament/loaded {:data {:players players
+                                                             :missing-players missing-players
+                                                             :rounds rounds
+                                                             :cobra-link cobra-link
+                                                             :tournament-name (:name data)}})))
+
+(defn wrap-with-to-handler
```

Comment:
This isn't needed because we have `wrap-tournament-auth-required` in `web.auth`. Better to remove the unnecessary `(if (:tournament-organizer user)` checks from these functions.